core/src/apps/mintlayer/sign_tx/signer.py

```diff
@@ -429,6 +483,58 @@ async def step3_serialize_inputs(self) -> Tuple[List[bytes], List[bytes]]:
                 encoded_inputs.append(encoded_inp)
                 # just add a \x00 for an empty Option as account command inputs don't have an UTXO
                 encoded_input_utxos.append(b"\x00")
+            elif inp.input.order_command:
+                x = inp.input.order_command
+
+                if x.conclude:
+                    ord = x.conclude
+                    encoded_inp = (
+                        mintlayer_utils.encode_conclude_order_v1_order_command_input(
+                            mintlayer_decode(
+                                ord.order_id, self.coininfo.prefixes.order
+                            ),
+                        )
+                    )
+                    encoded_inputs.append(encoded_inp)
+                    # just add a \x00 for an empty Option as order command inputs don't have an UTXO
+                    # TODO: fix when input commitments are ready
+                    encoded_input_utxos.append(b"\x00")
+                    continue
+                elif x.freeze:
+                    ord = x.freeze
+                    encoded_inp = (
+                        mintlayer_utils.encode_freeze_order_v1_order_command_input(
+                            mintlayer_decode(
+                                ord.order_id, self.coininfo.prefixes.order
+                            ),
+                        )
+                    )
+                    encoded_inputs.append(encoded_inp)
+                    # just add a \x00 for an empty Option as order command inputs don't have an UTXO
+                    # TODO: fix when input commitments are ready
```

Comment:
Btw, we won't actually commit to anything for FreezeOrder - committing to the initial balances is redundant and committing to current ones would allow a malicious actor to prevent order freezing by constantly sending "fill"s for small amounts.